Sources/SwiftProtobuf/ProtobufBinaryTypes.swift

```diff
@@ -36,6 +36,13 @@ public protocol ProtobufBinaryCodableType: ProtobufTypePropertiesBase {
 
     // Special interface for decoding a value of this type as a map value.
     static func decodeProtobufMapValue(decoder: inout ProtobufFieldDecoder, value: inout BaseType?) throws
+
+    /// Returns the number of byte required to encode `value` on the wire.
+    ///
+    /// Note that for length-delimited data (such as strings, bytes, and messages), the returned
+    /// size includes the space required for the length prefix. For messages, this is a subtle
+    /// distinction from the `serializedSize()` method, which does *not* include the length prefix.
+    static func encodedSize(of value: BaseType) throws -> Int
```

Comment:
I like how `encodedSize(of:)` reads, but there's a hard naming problem here.

There are several places where we need to distinguish:
* A full protobuf **field** (including leading tag)
* A protobuf **field body** (including size varint, where appropriate)
* A protobuf **field contents** (not including size varint) -- this is typically the same as the encoded object, so doesn't necessarily need a separate term.

It would be great to have a clear convention about how to name these that we can use throughout the library.

What do you think about `fieldBodySize(of:)` or `encodedFieldBodySize(of:)`?